### DISCUSSION TITLE
Search jobs by name

### DISCUSSION THREAD

**Discussion by KevinLeBonCMG** [felixmosh/bull-board#682] [Ideas]
Hello,

I have a lot of job with different name in a same queue and it's hard to find a specific job
Can you please add a search bar for jobs.

Thanks

**Comment by felixmosh** [comment]
Hi @KevinLeBonCMG
This feature is must requested one, but as I've mentioned in #361 there is no good solution for searching in bull nor bullmq...

If you have any interesting solution, I'll be more than happy to review a PR for this one.

**Reply by KevinLeBonCMG** [reply]
Yes after retrieving jobs from redis, you can filter on the display, job id or job.name and/or maybe params ?

**Reply by felixmosh** [reply]
This won't work since there is a pagination in place...
You can get a "page" which after filtering will result in an empty page ..

**Reply by KevinLeBonCMG** [reply]
And doing the process on server side before sending result ?

the web do the original request,
the server get the first chunk, filter, add to a result, then if result is less than "jobsPerPage", he get the next chunk, redo the filter and again and again, until the result is full, or got the last chunk ?

**Reply by felixmosh** [reply]
Yeah, this is really inefficient, therefore, as I've suggested in the linked issue, the underline lins (bull /bullmq) should add support for this...

**Reply by KevinLeBonCMG** [reply]
But maybe still more efficient than doing a ctrl+F and switching between 50 pages 😄

**Reply by felixmosh** [reply]
Yeah, that's right... The problem is that tomorrow someone will open an issue regarding slow response time (happened before), and I will need to solve it ;)

Let's push the request in bullmq lib
